python/lsst/daf/butler/remote_butler/server_models.py

```diff
@@ -31,31 +31,41 @@
     "CLIENT_REQUEST_ID_HEADER_NAME",
     "CollectionList",
     "DatasetTypeName",
-    "FindDatasetModel",
+    "FindDatasetRequestModel",
+    "FindDatasetResponseModel",
     "GetFileResponseModel",
 ]
 
 from typing import NewType
 
 import pydantic
-from lsst.daf.butler import SerializedDataId, SerializedDatasetRef
+from lsst.daf.butler import SerializedDataId, SerializedDatasetRef, SerializedTimespan
 from lsst.daf.butler.datastores.fileDatastoreClient import FileDatastoreGetPayload
 
 CLIENT_REQUEST_ID_HEADER_NAME = "X-Butler-Client-Request-Id"
 
 CollectionList = NewType("CollectionList", list[str])
+"""A list of search patterns for collection names.  May use glob
+syntax to specify wildcards."""
 DatasetTypeName = NewType("DatasetTypeName", str)
 
 
-class FindDatasetModel(pydantic.BaseModel):
+class FindDatasetRequestModel(pydantic.BaseModel):
     """Request model for find_dataset."""
 
     data_id: SerializedDataId
     collections: CollectionList
+    timespan: SerializedTimespan | None
     dimension_records: bool = False
     datastore_records: bool = False
 
 
+class FindDatasetResponseModel(pydantic.BaseModel):
```

Comment:
This change breaks backwards compatibility but we have breakage from other recent PRs that I was going to make new releases for already.

In theory I could have just changed the response type to be `SerializedDatasetRef | None`, but it's really useful to have a wrapper object for responses.  If you find out you need to add more data later, you can add it as an additional property without it being a backwards-incompatible change.